src/backends.rs

```diff
@@ -7,33 +7,59 @@
 //! The function MUST NOT ever write uninitialized bytes into `dest`,
 //! regardless of what value it returns.
 
+use core::mem::MaybeUninit;
+
+use crate::Error;
+
+/// If an external fallback _may_ be used, use it.
+/// If the fallback may not be used, the provided token trees will be included instead.
+///
+/// This is extracted into its own macro to allow using a fallback in multiple branches.
+/// For example, on unsupported WASI targets.
+#[allow(unused)]
+// May be unused if a fallback is not required.
+macro_rules! use_fallback_or {
+    ($($tt: tt)*) => {
+        cfg_if! {
+            if #[cfg(feature = "custom-fallback")] {
+                mod fallback;
+                pub use fallback::Implementation;
+            } else {
+                $($tt)*
+            }
+        }
+    }
+}
```

Comment:
This design (in addition to documentation on the `custom_fallback` feature flag) mostly solves one of my concerns: linker errors due to usage of an undefined function.